llvm/lib/Passes/PassBuilder.cpp

```diff
@@ -365,6 +365,34 @@ class TriggerVerifierErrorPass
   static StringRef name() { return "TriggerVerifierErrorPass"; }
 };
 
+// A pass requires all MachineFunctionProperties.
+// DO NOT USE THIS EXCEPT FOR TESTING!
+class RequireAllMachineFunctionPropertiesPass
+    : public MachinePassInfoMixin<RequireAllMachineFunctionPropertiesPass> {
+public:
+  PreservedAnalyses run(MachineFunction &MF, MachineFunctionAnalysisManager &) {
+    PropertyChanger PC(MF);
```

Comment:
can't we pull this out into the PassManager call sites for run, instead of inside the implementations of each of them?